scripts/generate_integration_test.py

```diff
@@ -50,49 +51,62 @@ def process_post(request: Request, file: IO) -> None:
+        and request.url == f"{RUNNER_ROOT_URL}/questionnaire/"
+    )
+    if has_journey_started:
+        survey_journey["in_progress"] = True
+        return
+
+    if is_recordable_survey_navigation(request):
         path = f'"{urlparse(request.url).path}"'
         file.write(generate_method_request(method="get", data=path))
 
-    if not survey_journey["started"]:
+    elif survey_journey["in_progress"]:
+        # ensure the request method is captured - allows us to record Runner GET navigation actions on the next pass through
         survey_journey["previous_request_method"] = request.method
 
-    if (
-        not survey_journey["previous_request_method"]
-        and survey_journey["started"]
-        and request.url == f"{RUNNER_ROOT_URL}/questionnaire/"
-    ):
-        survey_journey["started"] = False
-
 
 def process_launcher_request(request: Request) -> None:
-    if request.method == "GET":
-        if survey_journey["started"]:
-            # start of journey, so create a skeleton file using the schema name
-            survey_journey["schema_name"] = parse_qs(request.url)["schema_name"][0]
-            output["file_name"] = f"./scripts/{survey_journey['schema_name']}.py"
-
-            with open(output["file_name"], "w", encoding="utf-8") as file:
-                # Type ignore: schema_name is taken as string from query string
-                class_name = survey_journey["schema_name"].title().replace("_", "")  # type: ignore
-                file.write(
-                    TEST_TEMPLATE.format(
-                        class_name=class_name,
-                        function_name=survey_journey["schema_name"],
-                        schema_name=survey_journey["schema_name"],
-                    )
+    if request.method != "GET":
+        return
+
+    if not survey_journey["in_progress"]:
+        # start of journey, so create a skeleton file using the schema name
+        survey_journey["schema_name"] = parse_qs(request.url)["schema_name"][0]
+        output["file_name"] = f"./scripts/{survey_journey['schema_name']}.py"
+
+        with open(output["file_name"], "w", encoding="utf-8") as file:
+            # Type ignore: schema_name is taken as string from query string
+            class_name = survey_journey["schema_name"].title().replace("_", "")  # type: ignore
+            file.write(
+                TEST_TEMPLATE.format(
+                    class_name=class_name,
+                    function_name=survey_journey["schema_name"],
+                    schema_name=survey_journey["schema_name"],
                 )
-                logger.info(request)
-        else:
-            # capture launcher urls for sign-out, save etc
-            with open(output["file_name"], "a", encoding="utf-8") as file:
-                path = f'"{urlparse(request.url).path}"'
-                file.write(generate_method_request(method="get", data=path))
+            )
+            logger.info(request)
+    else:
```

Comment:
If you are always going to check both cases, i.e if true and when false, then avoid using inversing via `not` as the first check as it makes it less clear.

```py
if True:
    do_stuff()
else:
    do_other_stuff()
```

Over:

```py
if not True:
   do_other_stuff()
else:
   do_stuff()
```

Reply:
I think I tried to put the 'first' action of the file creation up front, but agree it's more difficult to read with inversion. Definitely makes that more readable, thanks 👍